gemini/sample-apps/e2e-gen-ai-app-starter-pack/tests/integration/evaluation/test_evaluation.py

```diff
@@ -0,0 +1,106 @@
+# Copyright 2024 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# pylint: disable=R0801
+
+import json
+
+from app.eval.utils import batch_generate_messages, generate_multiturn_history
+from app.patterns.custom_rag_qa.chain import chain
+from google.cloud import aiplatform
+from langchain_google_vertexai import ChatVertexAI
+import pandas as pd
+import pytest
+from vertexai.evaluation import CustomMetric, EvalTask
+import yaml
+
+
+@pytest.mark.asyncio
+async def test_multiturn_evaluation() -> None:
+    y = yaml.safe_load(open("tests/integration/evaluation/ml_ops_chat.yaml"))
+    df = pd.DataFrame(y)
+    df = generate_multiturn_history(df)
+
+    assert len(df) == 2
+    assert df["conversation_history"][0] == []
+    assert df["human_message"][0]["content"][0]["text"] == "Explain what's MLOps"
+    assert len(df["conversation_history"][1]) == 4
+    assert (
+        df["human_message"][1]["content"][0]["text"] == "How can I evaluate my models?"
+    )
+
+    scored_data = batch_generate_messages(df, chain)
```

Comment:
I thought that these functions don't exists anymore?

Reply:
When we spoke I thought the same, but then we'd have duplicated code in the notebook getting_started and here for the test...